### ISSUE TITLE
Possible UI update occurring on background thread in Framework

### ISSUE THREAD

**Issue by SoneeJohn** [DevMate/DevMateKit#31]
I attempted activation with the default UI provided by the framework and I got this stack track:

```

Main Thread Checker: UI API called on a background thread: -[NSControl stringValue]
PID: 42142, TID: 3534007, Thread name: (none), Queue name: com.apple.root.default-qos.overcommit, QoS: 21
Backtrace:
4   DevMateKit                          0x000000010093406b -[DMActivationStepController saveCurrentState:] + 114
5   DevMateKit                          0x000000010093aa60 -[DMActivationController allStepsStates] + 114
6   DevMateKit                          0x0000000100934acf -[DMActivationStepController performBackgroundActivation] + 401
7   Foundation                          0x00007fff49f63e48 __NSThread__start__ + 1197
8   libsystem_pthread.dylib             0x00000001016976b9 _pthread_body + 340
9   libsystem_pthread.dylib             0x0000000101697565 _pthread_body + 0
10  libsystem_pthread.dylib             0x0000000101696c55 thread_start + 13
2017-12-26 12:20:16.024327-0400 iTubeDownloader[42142:3534007] [reports] Main Thread Checker: UI API called on a background thread: -[NSControl stringValue]
PID: 42142, TID: 3534007, Thread name: (none), Queue name: com.apple.root.default-qos.overcommit, QoS: 21
Backtrace:
4   DevMateKit                          0x000000010093406b -[DMActivationStepController saveCurrentState:] + 114
5   DevMateKit                          0x000000010093aa60 -[DMActivationController allStepsStates] + 114
6   DevMateKit                          0x0000000100934acf -[DMActivationStepController performBackgroundActivation] + 401
7   Foundation                          0x00007fff49f63e48 __NSThread__start__ + 1197
8   libsystem_pthread.dylib             0x00000001016976b9 _pthread_body + 340
9   libsystem_pthread.dylib             0x0000000101697565 _pthread_body + 0
10  libsystem_pthread.dylib             0x0000000101696c55 thread_start + 13


```

**Comment by dimaty** [comment]
Fixed in v1.8

**Comment by SoneeJohn** [comment]
@dimaty I just noticed this bug occurring again while using the latest version of DevMate Kit 

```

Main Thread Checker: UI API called on a background thread: -[NSWindowController isWindowLoaded]
PID: 67049, TID: 2225347, Thread name: (none), Queue name: com.apple.root.default-qos, QoS: 0
Backtrace:
4   DevMateKit                          0x0000000100af7715 -[DMActivationController isRunning] + 23
5   DevMateKit                          0x0000000100af63c8 -[DMActivationController kevlarApplicationDidChangeActivationStatus:] + 51
6   CoreFoundation                      0x00007fff3b2f035f __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
7   CoreFoundation                      0x00007fff3b380c73 ___CFXRegistrationPost1_block_invoke + 63
8   CoreFoundation                      0x00007fff3b380308 _CFXRegistrationPost1 + 372
9   CoreFoundation                      0x00007fff3b2f89be ___CFXNotificationPost_block_invoke + 97
10  CoreFoundation                      0x00007fff3b260ce2 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1575
11  CoreFoundation                      0x00007fff3b260172 _CFXNotificationPost + 1351
12  Foundation                          0x00007fff3d92736b -[NSNotificationCenter postNotificationName:object:userInfo:] + 59
13  iTubeDownloader                     0x0000000100162935 __64-[NSApplication(hw9D4WiK1o) activateWithInfo:completionHandler:]_block_invoke + 2261
14  iTubeDownloader                     0x000000010019dd42 ___ZN60u6AI80eBXH8UDuAmCqHwXNVICkZ8GcNPdIuiFnHRuo7zybyplJgmGYMstcWC65ccoUZRztE0jm1eRibG58Toa6Dnwi2K0t82vEzay68nlNeKcYcp6U2W16jQ8ZsLwOHElP12NSDictionaryadP16dispatch_queue_sU13block_pointerFvaP7NSErrorE_block_invoke_2 + 50
15  libdispatch.dylib                   0x00000001010fb7b3 _dispatch_call_block_and_release + 12
16  libdispatch.dylib                   0x00000001010fc78f _dispatch_client_callout + 8
17  libdispatch.dylib                   0x00000001010fedf9 _dispatch_queue_override_invoke + 1038
18  libdispatch.dylib                   0x000000010111031d _dispatch_root_queue_drain + 334
19  libdispatch.dylib                   0x0000000101110d8f _dispatch_worker_thread2 + 127
20  libsystem_pthread.dylib             0x000000010118800d _pthread_wqthread + 220
21  libsystem_pthread.dylib             0x0000000101187eb3 start_wqthread + 15
```
